### PR TITLE
Feat/add claude code skill for code migration

### PR DESCRIPTION
**Pull request by bojtospeter**
# Pull Request Template

## Description
Add migration related claude code skill, so users can migrate with claude code end-to-end. The skill should be able to support deterministic (e.g. openrewrite) and agentic steps.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Test-only changes (no production code changes)

### Black-Box Testing Requirements
- [ ] Tests follow **black-box testing approach**: verify behavior through observable outputs (logs, C8 API queries, real-world skip scenarios)
- [ ] Tests **DO NOT** access implementation details (`DbClient`, `IdKeyMapper`, `..impl..` packages except logging constants)
- [ ] Architecture tests pass (`ArchitectureTest` validates these rules)

### Test Coverage
- [ ] Added tests for new functionality
- [ ] Updated tests for modified functionality
- [ ] All tests pass locally

## Architecture Compliance

Run architecture tests to ensure compliance:
```bash
mvn test -Dtest=ArchitectureTest
```

If architecture tests fail, refactor your tests to use:
- `LogCapturer` for log assertions
- `camundaClient.new*SearchRequest()` for C8 queries
- Real-World skip scenarios (e.g., migrate children without parents)

## Documentation
- [ ] Updated TESTING_GUIDELINES.md if adding new test patterns
- [ ] Added Javadoc comments for public APIs
- [x] Updated README if user-facing changes

## Checklist
- [ ] Code follows project style guidelines
- [x] Self-reviewed the code
- [ ] Added comments for complex logic
- [ ] No new compiler warnings
- [ ] Dependent changes have been merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,32 @@ A comprehensive toolkit for converting Camunda 7 code to Camunda 8. This module
 
 For detailed information, see the [Code Conversion README](./code-conversion/README.md).
 
+### Claude Code Skill
+
+If you use [Claude Code](https://claude.ai/code), you can install a `/migrate-c7-to-c8` skill that guides you through the full migration interactively — no copy-pasting prompts from docs.
```
Comment (maff) [inline README.md]: One thought on the framing: is this skill in any way specific to claude (outside of the installation/plugin)? If not, I'd suggest to refer to the skill as "(AI) Agent Skill" as documented on https://agentskills.io/.

As we explore the best ways to distribute/install I'd say it's fine to keep the installation section specific to claude.

Comment (bojtospeter) [reply README.md]: I don't know how general the support is for the format used by Claude's skills. I'm happy to make it general, if it really works with other agents out there.

Comment (maff) [reply README.md]: Skills started out in the Claude ecosystem but, but were moved to an independent spec which is now supported by many agents. Unless the content of the skill is very Claude specific I think they should equally work across different agents once installed.

**Comment by bojtospeter** [comment]
@copilot, the CI is failing, investigate and fix it

File: README.md (hunk 1 above)
Comment (bojtospeter) [reply README.md]: Rephrased it, suggesting claude as one of the options.

File: agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+- Update process instance startup patterns
+- Replace assertion methods: e.g. `isWaitingAt("id")` → `hasActiveElements("id")`
+- Update message correlation, timer handling, user task completion in tests
+- Reference: "Test assertions" patterns
+
+---
+
+### Approach C — Assessment only
+
+Present the assessment table from Step 3 with additional detail:
```
[the remark this answers is not in the document]
Comment (bojtospeter) [reply agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md]: fixed in a later commit

File: .claude-plugin/marketplace.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "name": "camunda",
+    "url": "https://github.com/camunda"
+  },
+  "plugins": [
+    {
+      "name": "camunda-migration",
+      "source": {
+        "source": "git-subdir",
+        "url": "https://github.com/camunda/camunda-7-to-8-migration-tooling.git",
+        "path": "agentic-migration-skills"
```
[the remark this answers is not in the document]
Comment (bojtospeter) [reply .claude-plugin/marketplace.json:14]: I've tested this approach with this repo:
https://github.com/bojtospeter/test-claude-skill/tree/main

@copilot please double check your recommendation

File: .claude-plugin/plugin.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "camunda-migration",
+  "version": "0.1.0",
+  "description": "Claude Code skills for migrating Camunda 7 projects to Camunda 8",
+  "author": {
+    "name": "camunda"
+  },
+  "license": "Camunda License 1.0",
+  "skills": "./agentic-migration-skills/skills/",
```
[the remark this answers is not in the document]
Comment (bojtospeter) [reply .claude-plugin/plugin.json:9]: I've tested this approach in https://github.com/bojtospeter/test-claude-skill/tree/main. @copilot please reconsider your recommendation.

File: README.md
Comment (venetrius) [inline README.md]: We could update this link as well, or remove it as it also present on line 123.
Currently point to next version https://docs.camunda.io/docs/next/guides/migrating-from-camunda-7
Line 123 docs point to current version: https://docs.camunda.io/docs/guides/migrating-from-camunda-7/

File: agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
+---
+name: migrate-c7-to-c8
+description: Migrate a Camunda 7 Java codebase to Camunda 8 using AI, OpenRewrite, or both
```
Comment (venetrius) [inline agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md]: From the [agentskills docs](https://agentskills.io/skill-creation/optimizing-descriptions#optimizing-skill-descriptions): 

> A skill only helps if it gets activated. The description field in your SKILL.md frontmatter is the primary mechanism agents use to decide whether to load a skill for a given task. An under-specified description means the skill won’t trigger when it should; an over-broad description means it triggers when it shouldn’t.

Given that this file intended for the AI agent to read specifing using AI as an option is not meaningfull `using AI, OpenRewrite, or both`


```suggestion
description: Use this when migrating or converting a Camunda 7 / camunda-bpm Java/Spring codebase to Camunda 8 — including JavaDelegates, ExternalTaskWorkers, ProcessEngine/RuntimeService client code, and application.properties/application.yaml with camunda.* keys.
```

File: agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
+---
```
Comment (venetrius) [inline agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md:1]: ❓ The skill is named migrate-c7-to-c8, but its scope is limited to code conversion (OpenRewrite recipes + AI). The other two tools in this repo — diagram-converter and data-migrator — aren't covered. Would it make sense to either rename the skill to reflect the narrower scope (e.g. migrate-c7-to-c8-code), or extend it to orchestrate all three?

File: agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+- `application.properties` / `application.yaml` with `camunda.*` keys → Config
+- `.bpmn` files with `camunda:` namespace attributes → BPMN (flag only — convert using the online tool, see below)
+
+After the table, present:
+- Total files to migrate
+- Overall complexity estimate
+- Whether OpenRewrite would help (JavaDelegates or ExternalTaskWorkers present?)
+- Any blockers requiring manual decision before starting
+
+Wait for user confirmation before proceeding.
```
Comment (venetrius) [inline agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md]: 💭 Worth being explicit about how to confirm: 
```suggestion
Use AskUserQuestion to wait for user confirmation before proceeding.
```

File: agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+Then stop — make no code changes.
+
+---
+
+## Step 4: Validation (always runs)
+
+1. **Compile**: `mvn compile` or `./gradlew compileJava` — fix all errors
+2. **Check for remaining C7 references**: Search for `org.camunda.bpm` imports — each is a missed migration
+3. **Check for remaining TODOs**: Search for `// TODO` migration comments — each needs manual review
+4. **Run tests**: `mvn test` or `./gradlew test` — fix failures
```
Comment (venetrius) [inline agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md:222]: 🤔 There are many edge cases where the code conversion requires manual steps. It's expected for the code not to compile cleanly. 
I'm on the fence whether I'd prefer the LLM to fix these autonomously or consult with me first. Same concern as the line above: `Search for // TODO migration comments — each needs manual review`.

**Review by venetrius — CHANGES_REQUESTED** [review]
Cool stuff! 
I added a few comments, questions. 
I did not test the skill myself yet. Do you have recommended steps for testing?

**Comment by bojtospeter** [comment]
> Do you have recommended steps for testing?

@venetrius you can test it by installing the skill with:
symlink the skill into your user skills dir so edits in the repo are picked up live:
`ln -s "$PWD/agentic-migration-skills/skills/migrate-c7-to-c8" ~/.claude/skills/migrate-c7-to-c8`
  Then restart Claude Code. The skill appears in Skill tool's available list and can be invoked as /migrate-c7-to-c8.

File: README.md
Comment (bojtospeter) [reply README.md]: taken care of, removed duplication and consistent version usage (no `next`)

File: agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md (hunk 5 above)
Comment (bojtospeter) [reply agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md]: done, thanks for the recommendation

File: agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md (hunk 6 above)
Comment (bojtospeter) [reply agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md:1]: renamed. When we will have more skill coverage, we should create more skills (e.g. diagram conversion), instead of making one skill which can do everything.

File: agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md (hunk 7 above)
Comment (bojtospeter) [reply agentic-migration-skills/skills/migrate-c7-to-c8/SKILL.md]: implemented, thanks

File: agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md (hunk 8 above)
Comment (bojtospeter) [reply agentic-migration-skills/skills/migrate-c7-to-c8-code/SKILL.md:222]: added another instruction to consult the user on edge-cases.

**Comment by bojtospeter** [comment]
@venetrius I've addressed your comments, please take a look and take the skill to a test-drive.

**Review by venetrius — APPROVED** [review]
👍 Looks good to me!
Thanks implementing my feedback. 
Did some experimenting. Used the skill on the `camunda-bpm-platform-maintenance/examples/invoice` module. It was able to set up and execute the code-conversion recipes, communicated on blockers and managed to reach a state where the code compiles. Really impressive.
